easybuild/tools/module_generator.py

```diff
@@ -213,18 +213,10 @@ def _filter_paths(self, key, paths):
             return paths
 
         added_paths = self.added_paths_per_key.setdefault(key, set())
-        # paths can be a string
-        if isinstance(paths, str):
-            if paths in added_paths:
-                filtered_paths = None
-            else:
-                added_paths.add(paths)
-                filtered_paths = paths
-        else:
-            # Coerce any iterable/generator into a list
-            if not isinstance(paths, list):
-                paths = list(paths)
-            filtered_paths = [x for x in paths if x not in added_paths and not added_paths.add(x)]
+        # Coerce any iterable/generator into a list
+        if not isinstance(paths, list):
+            paths = list(paths)
+        filtered_paths = [x for x in paths if x not in added_paths and not added_paths.add(x)]
         if filtered_paths != paths:
             removed_paths = paths if filtered_paths is None else [x for x in paths if x not in filtered_paths]
             print_warning("Suppressed adding the following path(s) to $%s of the module as they were already added: %s",
```

Comment:
In the below line you can use `', '.join(removed_paths)` so you don't need to change the tests and improve the output